src/messages.rs

```diff
@@ -33,15 +33,15 @@ pub struct Message {
     pub header: Header,
     pub payload: Payload,
 
-    pub metadata: TransportData,
+    pub metadata: Option<TransportData>,
```

Comment:
Keep in mind that we might again change `TransportData`  later on when adding new fields wrt `KadcastSendToMany` and `KadcastSendToOne`.

Reply:
I agree, but so far this field is used to map the incoming kadcast metadata. So it should make sense to have it None if message are not generated by the kadcast receiver.

If we'll need in the future to add some other fields, we'll do it

Reply:
Makes sense!